hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MutableQuantiles.java

```diff
@@ -130,11 +129,11 @@ public MutableQuantiles() {}
   public synchronized void snapshot(MetricsRecordBuilder builder, boolean all) {
     if (all || changed()) {
       builder.addGauge(numInfo, previousCount);
-      for (int i = 0; i < quantiles.length; i++) {
+      for (int i = 0; i < getQuantiles().length; i++) {
```

Comment:
You probably need to extract `getQuantiles()`, otherwise you may end up with two versions.

Reply:
updated